punishments.js

```diff
@@ -1114,7 +1114,16 @@ Punishments.checkIp = function (user, connection) {
 		}
 	}
 
-	Dnsbl.reverse(ip).then(host => {
+	Dnsbl.reverse(ip).catch(e => {
+		if (e.code === 'EINVAL') {
+			if (connection.user && !connection.user.locked && !connection.user.autoconfirmed) {
```

Comment:
Why `connection.user` and not `user`?

Reply:
I was using `connection.user` because the rest of `Punishments.checkIp` handle semilocks the same way. The only reason I can think of is to make the async parts of this semilock the right user if `User#merge` can be called before they run, but if `Connection#onDisconnect` can as well, this would be preventing the user from getting semilocked in that case.

There's probably a better way to deal with those races if they happen at all, but if they don't, it'd probably be a better idea to use `user` instead and make `User#merge` handle semilocks properly. It's something I'll need to test at some point

Reply:
I'd just explicitly set `user = connection.user` so people know that's not a bug. Maybe even add a comment to that effect.